### PR TITLE
Update st2-pack-install command so it allows users to install packs from local git repos in a detached head state

### PR DESCRIPTION
**Pull request by Kami**
This pull request updates ``st2-pack-install`` command to allow users install packs from a local directory on disk (`file:///<path>`) which are in a detached head state - aka user already checked out a specific commit / revision.

Fixes https://github.com/StackStorm/st2packs-dockerfiles/issues/4.

## TODO

- [x] Tests
- [x] Changelog entry

### CODE REVIEW

**Comment by AndyMoore** [comment]
```Step 9/34 : RUN /usr/bin/st2-pack-install file:///packs/consul
 ---> Running in 0b73de203a99
2018-09-28 10:34:11,670 INFO [-] Installing pack "file:///packs/consul"
2018-09-28 10:34:12,077 INFO [-] Successfully installed pack "consul"
2018-09-28 10:34:12,077 INFO [-] Setting up virtualenv for pack "consul"
2018-09-28 10:34:12,078 INFO [-] Virtualenv path "/opt/stackstorm/virtualenvs/consul" doesn't exist
2018-09-28 10:34:16,297 INFO [-] Successfully set up virtualenv for pack "consul"
Removing intermediate container 0b73de203a99
```

👍

File: st2common/st2common/util/pack_management.py [hunk 1]
```diff
@@ -151,7 +151,22 @@ def clone_repo(temp_dir, repo_url, verify_ssl=True, ref='master'):
     # because we want the user to work with the repo in the
     # future.
     repo = Repo.clone_from(repo_url, temp_dir)
-    active_branch = repo.active_branch
+
+    is_local_repo = repo_url.startswith('file:///')
```
Comment (armab) [inline st2common/st2common/util/pack_management.py]: File URI can be with just 2 slashes in it like `file://localhost/some/local/dir`, see https://en.wikipedia.org/wiki/File_URI_scheme

Comment (Kami) [reply st2common/st2common/util/pack_management.py]: Yeah, I don't think rest of our pack management code code supports that though :)

Comment (Kami) [reply st2common/st2common/util/pack_management.py]: You are correct - we didn't have any tests cases so I assume it's not supported / doesn't work.

Will make a change and add a test case.

**Pull request merged**